cpp/search/search.cpp

```diff
@@ -2003,7 +2004,11 @@ void Search::initNodeNNOutput(
   }
   if(isRoot && searchParams.rootNumSymmetriesToSample > 1) {
     vector<shared_ptr<NNOutput>> ptrs;
+    std::array<int, NNInputs::NUM_SYMMETRY_COMBINATIONS> symmetryIndexes;
+    std::iota(symmetryIndexes.begin(), symmetryIndexes.end(), 0);
     for(int i = 0; i<searchParams.rootNumSymmetriesToSample; i++) {
+      std::swap(symmetryIndexes[i], symmetryIndexes[thread.rand.nextInt(i,NNInputs::NUM_SYMMETRY_COMBINATIONS-1)]);
+      nnInputParams.symmetry = symmetryIndexes[i];
       bool skipCacheThisIteration = skipCache || i > 0; //Skip cache on subsequent iterations to get new random draws for orientation
```

Comment:
Seems like this is going to be buggy because if the cache already has this position, on iteration 0 we won't get the symmetry we asked for, right?

It used to be okay to trust the cache for iteration 0 because we were sampling without replacement, the cached value was as good as any other arbitrary draw for the first random symmetry. But now that we have a specific array we have shuffled and that we are relying on getting an exact specific symmetry, we can no longer use the cache for *any* of the iterations, right?

Seems like this is going to be a very slight extra performance/latency cost on the root node if we can't use the cache even on iteration 0 (because the root node also blocks any possible multithreading to search the rest of the tree too). It makes this feature somewhat more expensive than before. But maybe it's not too big a deal.

Reply:
Cache always skipped in this part of the code now.